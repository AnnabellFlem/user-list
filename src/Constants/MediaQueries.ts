export const MEDIA_QUERY_MOBILE = {
  minWidth: '601px',
}

export const MEDIA_QUERY_TABLET = {
  minWidth: '921px',
}
