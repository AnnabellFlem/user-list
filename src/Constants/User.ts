export const MAIN_EDIT_FORM = 'main-form'
export const PER_PAGE = 6
