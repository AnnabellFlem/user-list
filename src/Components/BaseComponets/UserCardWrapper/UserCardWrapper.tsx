import React, { useState } from 'react'
import UserCardForm from '../UserCardForm'
import UserCard from '../UserCard'
import { User } from '../../../Types/User'
import { MAIN_EDIT_FORM_TYPE } from '../../../Constants/User'
import { initialUserValues } from '../../../Utils/getInitialUserData'

type UserCardWrapperProps = Partial<{
  user: User
  initUserValues: User
  editFormType: string
}>

const UserCardWrapper: React.FC<UserCardWrapperProps> = ({
  user,
  editFormType,
}) => {
  const [userForm, setUserForm] = useState<boolean>()

  const handleEditClick = () => {
    setUserForm(true)
  }

  const handleSaveClick = () => {
    setUserForm(false)
  }

  const handleCancelClick = () => {
    setUserForm(false)
  }

  const handleCreateClick = () => {
    setUserForm(true)
  }

  const isMainEditForm = (formType: string) => {
    return editFormType === formType
  }

  const renderUserCard = () => {
    if (userForm) {
      return (
        <li className="user">
          <UserCardForm
            user={user}
            handleCreateClick={handleCreateClick}
            handleCancelClick={handleCancelClick}
          />
        </li>
      )
    } else if (isMainEditForm(MAIN_EDIT_FORM_TYPE)) {
      return <UserCardForm isMainEditForm handleSaveClick={handleSaveClick} />
    } else {
      return (
        <UserCard
          user={user || initialUserValues}
          handleClick={handleEditClick}
        />
      )
    }
  }

  return <>{renderUserCard()}</>
}

export default UserCardWrapper
