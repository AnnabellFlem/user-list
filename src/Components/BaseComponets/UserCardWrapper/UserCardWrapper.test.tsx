import React from 'react'
import { shallow } from 'enzyme'
import UserCardWrapper from './UserCardWrapper'
import { MAIN_EDIT_FORM_TYPE } from '../../../Constants/User'

const mockedData = {
  id: 'id',
  name: 'name',
  surname: 'surname',
  email: 'email',
  phone: 'phone',
  birthday: 'birthday',
  credit: 'credit',
}

describe('UserCardWrapper test', () => {
  let wrapper: any

  beforeAll(() => {
    wrapper = shallow(<UserCardWrapper user={mockedData} />)
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Component should render UserCard', () => {
    expect(wrapper.find('UserCard')).toHaveLength(1)
  })

  it('Component should render UserFormCard', () => {
    wrapper = shallow(
      <UserCardWrapper user={mockedData} editFormType={MAIN_EDIT_FORM_TYPE} />,
    )
    expect(wrapper.find('UserCardForm')).toHaveLength(1)
  })
})
