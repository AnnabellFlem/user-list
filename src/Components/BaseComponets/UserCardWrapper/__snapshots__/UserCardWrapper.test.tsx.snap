// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCardWrapper test should to match snapshot 1`] = `ShallowWrapper {}`;
